aiida/cmdline/commands/cmd_import.py

```diff
@@ -108,36 +212,54 @@ def cmd_import(archives, webpages, group, extras_mode_existing, extras_mode_new,
             except Exception:
                 echo.echo_error('an exception occurred while trying to discover archives at URL {}'.format(webpage))
                 echo.echo(traceback.format_exc())
-                click.confirm('do you want to continue?', abort=True)
+                if not non_interactive:
+                    click.confirm('do you want to continue?', abort=True)
             else:
                 echo.echo_success('{} archive URLs discovered and added'.format(len(urls)))
                 archives_url += urls
 
+    # Preliminary sanity check
     if not archives_url + archives_file:
         echo.echo_critical('no valid exported archives were found')
 
+    # Import initialization
+    import_opts = {
+        "file_to_import": '',
+        "archive": '',
+        "group": group,
+        "migration": migration,
+        "extras_mode_existing": extras_mode_existing,
+        "extras_mode_new": extras_mode_new,
+        "comment_mode": comment_mode,
+        "non_interactive": non_interactive
+    }
+
+    # Import local archives
     for archive in archives_file:
 
         echo.echo_info('importing archive {}'.format(archive))
 
-        try:
-            import_data(
-                archive,
-                group,
-                extras_mode_existing=ExtrasImportCode[extras_mode_existing].value,
-                extras_mode_new=extras_mode_new,
-                comment_mode=comment_mode)
-        except exceptions.IncompatibleArchiveVersionError as exception:
-            echo.echo_critical('{} cannot be imported: {}'.format(archive, exception))
-        except Exception:
-            echo.echo_error('an exception occurred while importing the archive {}'.format(archive))
-            echo.echo(traceback.format_exc())
-            click.confirm('do you want to continue?', abort=True)
-        else:
-            echo.echo_success('imported archive {}'.format(archive))
+        # Initialization
+        import_opts['archive'] = archive
+        import_opts['file_to_import'] = import_opts['archive']
+
+        # First attempt to import archive
+        migrate_archive = _try_import(first=True, **import_opts)
+
+        # Migrate archive if needed and desired
+        if migrate_archive:
+
+            echo.echo_info("migrating archive {}".format(archive))
```

Comment:
let's move that inside `_migrate_archive`

Reply:
Done